scripts/build_renewable_profiles.py

```diff
@@ -263,21 +267,77 @@
         #         for geom in hydrobasins.geometry
         #     ]
         # ]  # exclude hydrobasins shapes that do not intersect the countries of interest
-        resource["plants"] = regions.rename(columns={"x": "lon", "y": "lat"}).loc[
+
+        resource["plants"] = regions.rename(
+            columns={"x": "lon", "y": "lat", "country": "countries"}
+        ).loc[
             [
                 # select busbar whose location (p) belongs to at least one hydrobasin geometry
-                any(hydrobasins.geometry.intersects(p))
-                for p in gpd.points_from_xy(regions.x, regions.y, crs=regions.crs)
+                # if extendable option is true, all buses are included
+                # otherwise only where hydro powerplants are available are considered
+                (
+                    (
+                        config.get("extendable", False)
+                        | (bus_id in hydro_ppls.bus.values)
+                    )
+                    & any(hydrobasins.geometry.intersects(p))
+                )
+                for (p, bus_id) in zip(
+                    gpd.points_from_xy(regions.x, regions.y, crs=regions.crs),
+                    regions.index,
+                )
             ],
-            ["lon", "lat"],
+            ["lon", "lat", "countries"],
         ]  # TODO: filtering by presence of hydro generators should be the way to go
+        resource["plants"]["installed_hydro"] = [
+            True if (bus_id in hydro_ppls.bus.values) else False
+            for bus_id in resource["plants"].index
+        ]
+
+        # check if normalization field belongs to the settings and it is not false
+        if ("normalization" in resource) & (resource["normalization"] is not bool):
+
+            normalization = resource.pop("normalization")
+
+            if normalization == "hydro_capacities":
+                hydro_stats = (
+                    pd.read_csv(
+                        snakemake.input.hydro_capacities,
+                        comment="#",
+                        na_values="-",
+                        index_col=0,
+                    )
+                    .rename({"Country": "countries"}, axis=1)
+                    .set_index("countries")
+                )
+                hydro_prod_by_country = hydro_stats[hydro_stats.index.isin(countries)][
+                    ["InflowHourlyAvg[GWh]"]
+                ].transpose()
+
+                year_start = 2000
+                year_end = 2020
+                range_years = range(year_start, year_end + 1)
+
+                normalize_using_yearly = pd.concat(
+                    [hydro_prod_by_country] * len(range_years), ignore_index=True
+                )
+                normalize_using_yearly.index = pd.Index(range_years)
+
+                resource["normalize_using_yearly"] = normalize_using_yearly
+            logger.info(f"Hydro normalization mode {normalization}")
```

Comment:
`if normalization == "eia":` is missing?

Reply:
Currently yes, I tried to investigate its use using the pypsa-eur-like approach, however I couldnt' find the exact platform to have the data in the same format. I found the format, but while downloading the data a different dataformat was provided, hence I first used the hydro_capacities data that I think it is a nice-have anyawy
I can ask them.

Reply:
Just saw this issue: https://github.com/PyPSA/pypsa-eur/issues/324

Reply:
Implemented a preliminry eia methodology based on pypsa-eur